File: EP-1/Python/array_to_ll.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def arr_to_ll(arr):
    """Convert array to linked list"""
    if not arr:
        return None
    
    head = Node(arr[0])
    mover = head
    
    for i in range(1, len(arr)):
        temp = Node(arr[i])
        mover.next = temp
        mover = temp
    
    return head

def print_ll(head):
    """Print linked list and return count"""
    temp = head
    cnt = 0
    while temp:
        cnt += 1
        print(temp.data, end=" ")
        temp = temp.next
    return cnt

if __name__ == "__main__":
    arr = [12, 13, 1, 3, 5]
    head = arr_to_ll(arr)
    cnt = print_ll(head)
    print(f"\nTotal number of nodes: {cnt}")

File: EP-1/Java/ArrayToLL.java
public class ArrayToLL {
    
    static class Node {
        int data;
        Node next;
        
        Node(int data) {
            this.data = data;
            this.next = null;
        }
    }
    
    public static Node arrToLL(int[] arr) {
        if (arr.length == 0) return null;
        
        Node head = new Node(arr[0]);
        Node mover = head;
        
        for (int i = 1; i < arr.length; i++) {
            Node temp = new Node(arr[i]);
            mover.next = temp;
            mover = temp;
        }
        return head;
    }
    
    public static int print(Node head) {
        Node temp = head;
        int cnt = 0;
        while (temp != null) {
            cnt++;
            System.out.print(temp.data + " ");
            temp = temp.next;
        }
        return cnt;
    }
    
    public static void main(String[] args) {
        int[] arr = {12, 13, 1, 3, 5};
        Node head = arrToLL(arr);
        int cnt = print(head);
        System.out.println("\nTotal number of nodes: " + cnt);
    }
}
